### PR TITLE
Preserve css-classes on copy and paste of lists. v2

### PR DESCRIPTION
**Pull request by elbird**
Open new Pull-Request because feature branch was broken. (old PR #1615)
When copying lists which are styled with custom classes (via the list plugin), the custom classes where not preserved on paste. To fix this
issue the generic content handler was rewritten: plugins can now register a PluginContentHandler and are able to define which parts of a
pasted content should be handled by a PluginContentHandler instead of the GenericContentHandler. For the list plugin a PluginContentHandler
was implemented which preserves the configured classes and also adds the configured default classes.

### CODE REVIEW

**Comment by elbird** [comment]
Fixed all tests which failed because of the current changes. But there are still Test failing also fail in hotfix-branch - see SUP-6826

File: doc/guides/source/writing_plugins.textile [hunk 1]
```diff
@@ -299,3 +299,47 @@ dependent plugin is actually loaded, and outpus an error message if not.
...
+Plugins can register their own ContentHandler to customize the behavior of
+the GenericContentHandler for certain Html-Elements. 
+
+To register a PluginContentHandler, the Plugin must implement a "getPluginContentHandler" 
+method. The method must return a map which maps a CSS-selector to a PluginContentHandler-function. 
+The PluginContentHandler-function will be called every time the GenericContentHandler encounters 
+an HTML-Element which matches the given CSS-selector. The function will get the current element, 
+options and editable (similiar to a "regular" ContentHandler) and can then perform DOM-manipulation 
+on the current element. A PluginContentHandler-function should only change the the element itself 
+and not it's children or parents.
```
Comment (npomaroli) [inline doc/guides/source/writing_plugins.textile]: its

File: src/lib/aloha/plugincontenthandlermanager.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                    Aloha.Log.warn(plugin, 'Cannot register a handler that is not a function!');
+                    return
+                }
+                handlers[selector].push(selectorHandlers[i]);
+            }
+        });
+    }
+
+    /**
+     * @param {jQuery.<HTMLElement>}
```
Comment (npomaroli) [inline src/lib/aloha/plugincontenthandlermanager.js]: @return

File: src/lib/util/contenthandler.js [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+			// normalize these table cells to contain actual white space
+			// instead.
+			if (isProppedParagraph($elem.html())) {
+				$elem.html('&nbsp;');
+			}
+		}
+	}
+
+	/**
+	 * Remove all comments
```
Comment (npomaroli) [inline src/lib/util/contenthandler.js]: Documentation and correct @param

File: src/lib/util/contenthandler.js [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+	 * @param content
+	 */
+	function removeComments($elem) {
+		if ($elem[0].nodeType === 8) {
+			$elem.remove();
+		}
+	}
+
+	/**
+	 * Remove styles
```
Comment (npomaroli) [inline src/lib/util/contenthandler.js]: Documentation and correct @param

File: src/lib/util/contenthandler.js [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+		$elem.removeAttr('style').removeClass();
+	}
+	
+
+	/**
+	 * Remove some unwanted tags from content paste
+	 * @param {jQuery.<HTMLElement>} $elem The HTMLElement which will be
+	 *                         		handled.
+	 * @param selector the tags to unwrap
+	 * @return boolean false when the element
```
Comment (npomaroli) [inline src/lib/util/contenthandler.js]: what?

File: src/lib/util/contenthandler.js [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+			typeof Aloha.settings.contentHandler.handler.generic.transformFormattings !== 'undefinded' &&
+			!Aloha.settings.contentHandler.handler.generic.transformFormattings) {
+			return false;
+		}
+		return true;
+	}	
+	/**
+	 * Do generic cleanup recursivley
+	 */
+	function doGenericCleanupRecursivley($elements) {
```
Comment (npomaroli) [inline src/lib/util/contenthandler.js]: doGenericCleanupRecursively (all occurrances)

File: src/plugins/common/contenthandler/lib/genericcontenthandler.js [hunk 7]
```diff
@@ -28,221 +28,79 @@ define([
...
-
+		/**
+		 * Apply the generic and the plugin content handler to HTMLElements and their children.
+		 * 
+		 * @param {jQuery.<Array.<HTMLElement>>} $element The HTMLElements which will be
+		 *                         		handled.
+		 * @param {object} options Used to filter limit which content handlers
+		 *                         should be used.
+		 * @param {Aloha.Editable} editable The editable whose content is currently being handled.
+		 */
```
Comment (npomaroli) [inline src/plugins/common/contenthandler/lib/genericcontenthandler.js]: add @param transformFormatting

File: src/plugins/common/list/lib/list-plugin.js [hunk 8]
```diff
@@ -496,6 +520,63 @@ define([
...
+					// cleanup empty class attributes
+					if (!$elem.attr('class')) {
+						$elem.removeAttr('class');
+					}
+					// return false to not apply other handlers and skip the generic cleanup on this element
+					return false;
+				},
+				'li,dt,dd': function ($elem, $options, editable) {
+					// if li,dt,dd elements are not properly nested
+					// we have to remove them entirely
```
Comment (npomaroli) [inline src/plugins/common/list/lib/list-plugin.js]: we unwrap the contents

File: src/test/unit/listcontenthandler.html [hunk 9]
```diff
@@ -0,0 +1,171 @@
+<!DOCTYPE html>
```
Comment (npomaroli) [inline src/test/unit/listcontenthandler.html]: Add this test to org.alohaeditor.SimpleTest.suite()

**Pull request merged**